### PR TITLE
Add a timeout for scan-files

### PR DESCRIPTION
**Pull request by smcmurtry**
# Summary | Résumé

This PR adds a timeout for the virus scan. This will protect Notify so it can continue to send in the event that scan-files goes down or is taking a really long time. If the virus scan has not completed by `SCAN_TIMEOUT_SECONDS` (currently set to 10 minutes) after the file object was created in S3, then return a 200 code along with the status `scan_timed_out` so the email attachment can be sent anyway. 

Here is the [updated decision tree diagram](https://github.com/cds-snc/notification-adr/blob/feat/malicious-content-scanning/records/2023-01-24.malicious-content-scanning.md#duplicate-s3-bucket) covering this logic.

# Test instructions | Instructions pour tester la modification

I can't think of a way to test this without either a) taking scan-files down, or b) modifying the code in this PR to return an in-progress. So here are the instructions for b):
1. check out this branch
2. add the following code to app/utils/store.py line 112: `raise ScanInProgressError("Content scanning is in progress")
`. This will simulate a scan that either never starts or does not complete
3. Change the timeout to 2 minutes so you don't have to wait so long
4. run with `make run`
5. set notification-api to point to your local dd-api, with `DOCUMENT_DOWNLOAD_API_HOST=http://host.docker.internal:7000`, and run locally
6. send an email w/ an attachment via your local API.
7. The email should not send for the first 2 minutes (the timeout duration), and then it should be sent successfully at some point after 2 minutes elapses.
8. verify that you see the dd-api log: `Scan timed out for document: ...`

### CODE REVIEW

File: app/download/views.py [hunk 1]
```diff
@@ -120,6 +121,19 @@ def check_scan_verdict(service_id, document_id):
...
+        )
+        if age_seconds > SCAN_TIMEOUT_SECONDS:
+            current_app.logger.info(
+                "Scan timed out for document: {}".format(e),
+                extra={
+                    "service_id": service_id,
+                    "document_id": document_id,
+                },
+            )
+            return jsonify(scan_verdict="scan_timed_out"), 200
```
Comment (jimleroyer) [inline app/download/views.py:135]: Will that be a `200` http status code if the scan timed out? Do we want to send something more specific to send to the API?

**Review by jimleroyer — COMMENTED** [review]
Nice additions!

Not approving yet as I know Jumana and/or Steve will test locally first, I'll let them follow through.

File: app/download/views.py (hunk 1 above)
Comment (smcmurtry) [reply app/download/views.py:135]: I chose `200` because I wanted this to work with the existing logic in notification-api and to result in a send, so we would need to change the logic over there if we return a different error code: https://github.com/cds-snc/notification-api/blob/main/app/delivery/send_to_providers.py#L158

The server will send the json data `{"scan_verdict": "scan_timed_out"}` in the response so I figured we could read that if we want to detect this in notification-api in the future.

**Review by sastels — APPROVED** [review]
Tested locally.

- made changes as suggested:
  - always return "scan in progress"
  - set timeout to 2 minutes

Tl;dr worked as expected:
 - first send attempt aborted  since scanning in progress.
 - 5 minutes later second attempt sent successfully, complaining that the scan timed out.

Details:

Api:
[2023-02-20 19:22:22,348: INFO/ForkPoolWorker-2] Malware scan in progress, could not download files for notification.id: c7566643-404c-4906-9bed-65e3af887105
[2023-02-20 19:22:22,349: WARNING/ForkPoolWorker-2] The exception is MalwareScanInProgressException()
[2023-02-20 19:22:22,349: WARNING/ForkPoolWorker-2] RETRY 0: Email notification c7566643-404c-4906-9bed-65e3af887105 failed
[2023-02-20 19:22:22,444: INFO/ForkPoolWorker-2] Task deliver_email[6292b7ae-3b6f-44fd-851b-b97d8cbe8cee] retry: Retry in 300s

Dd-api:
2023-02-20T19:22:22 Name app INFO None "Scan in progress, refused to download document: Content scanning is in progress" [in /workspaces/notification-document-download-api/app/download/views.py:137]


5 minutes later

Dd-api:
2023-02-20T19:27:22 Name app INFO None "Scan timed out for document: Content scanning is in progress" [in /workspaces/notification-document-download-api/app/download/views.py:128]


Api:
[2023-02-20 19:27:22,838: INFO/ForkPoolWorker-2] scan_verdict for document_id b4b10695-d9fd-4fe0-968a-fa7ece75aefa is {'scan_verdict': 'scan_timed_out'}


Email then sent successfully with attachment

File: app/download/views.py (hunk 1 above)
Comment (jimleroyer) [reply app/download/views.py:135]: The status code would be an option more aligned with HTTP (or RESTful), as no parsing would be required. There is even a status code dedicated for it, i.e. [408](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/408).

Comment (jimleroyer) [reply app/download/views.py:135]: `200` usually always means a success, but it's not a success here, so that could be misleading, albeit only to developers of Notify as this is an internal call.

Comment (jimleroyer) [reply app/download/views.py:135]: Also (sorry to add more), and that could be more serious, if some library interprets this code, they will assume it's a success. We might not be affected now but we could in the future (API calls reporting success would be all `200` for example -- but that should be an error).

Comment (smcmurtry) [reply app/download/views.py:135]: Fair enough, I will modify the status code in 2 follow up PRs, one on the dd-api and one on notification-api.
